controller/service.rst

```diff
@@ -144,227 +71,68 @@ the route ``_controller`` value:
+---------------------
 
-    If your controller implements the ``__invoke()`` method, you can simply
-    refer to the service id (``AppBundle\Controller\HelloController`` or
-    ``app.hello_controller`` for example).
+If your controller implements the ``__invoke()`` method - popular with the
+Action-Domain-Response (ADR) pattern, you can simply refer to the service id
+(``AppBundle\Controller\HelloController`` or ``app.hello_controller`` for example).
 
 Alternatives to base Controller Methods
 ---------------------------------------
 
-When using a controller defined as a service, it will most likely not extend
-the base ``Controller`` class. Instead of relying on its shortcut methods,
-you'll interact directly with the services that you need. Fortunately, this is
-usually pretty easy and the base `Controller class source code`_ is a great
-source on how to perform many common tasks.
+When using a controller defined as a service, you can still extend any of the
+:ref:`normal base controller <the-base-controller-class-services>` classes and
+use their shortcuts. But, you don't need to! You can choose to extend *nothing*,
+and use dependency injection to access difference services.
 
-For example, if you want to render a template instead of creating the ``Response``
-object directly, then your code would look like this if you were extending
-Symfony's base controller::
-
-    // src/AppBundle/Controller/HelloController.php
-    namespace AppBundle\Controller;
-
-    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
-
-    class HelloController extends Controller
-    {
-        public function indexAction($name)
-        {
-            return $this->render(
-                'AppBundle:Hello:index.html.twig',
-                array('name' => $name)
-            );
-        }
-    }
-
-If you look at the source code for the ``render()`` function in Symfony's
-`base Controller class`_, you'll see that this method actually uses the
-``templating`` service::
-
-    public function render($view, array $parameters = array(), Response $response = null)
-    {
-        return $this->container->get('templating')->renderResponse($view, $parameters, $response);
-    }
+The base `Controller class source code`_ is a great way to see how to performance
+simple tasks. For example, ``$this->render()`` is usually used to render a Twig
+template and return a Response. But, you can also do this directly:
 
 In a controller that's defined as a service, you can instead inject the ``templating``
 service and use it directly::
 
     // src/AppBundle/Controller/HelloController.php
     namespace AppBundle\Controller;
 
-    use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
     use Symfony\Component\HttpFoundation\Response;
 
     class HelloController
     {
-        private $templating;
+        private $twig;
 
-        public function __construct(EngineInterface $templating)
+        public function __construct(\Twig_Environment $twig)
```

Comment:
I agree... and - the new system makes it more obvious when a class is poorly-named in a library we use. Before, we could "hide" it behind our Symfony-specific service ids. I think it's something that needs to be improved in the underlying libraries (sometimes, like when the underlying classes are from Symfony - I'm thinking about security) - we can make that change ourselves :). For Twig, it might be interesting to introduce a new `Twig` or `TwigRenderer` class that only has a sub-set of the public functions that `Twig_Environment` has.